File: src/main/java/ac/kr/deu/connect/luck/image/ImageRestController.java
package ac.kr.deu.connect.luck.image;

import io.swagger.v3.oas.annotations.Parameter;
import io.swagger.v3.oas.annotations.tags.Tag;
import lombok.AllArgsConstructor;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestPart;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.multipart.MultipartFile;

@RestController
@Tag(name = "09-Image", description = "Image API")
@RequestMapping("/api/image")
@AllArgsConstructor
public class ImageRestController {

    private final ImageUploader imageUploader;

    @PostMapping(consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
    public ImageUrlResponse uploadImage(
            @Parameter(description = "multipart/form-data 형식의 이미지 리스트를 input으로 받습니다. 이때 key 값은 image 입니다.")
            @RequestPart("image") MultipartFile multipartFile) {
        return new ImageUrlResponse(imageUploader.uploadImage(multipartFile).getData().getUrl());
    }

}

File: src/main/java/ac/kr/deu/connect/luck/common/controller/ImageRestController.kt
package ac.kr.deu.connect.luck.common.controller

import ac.kr.deu.connect.luck.common.storage.StorageService
import io.swagger.v3.oas.annotations.media.Schema
import io.swagger.v3.oas.annotations.tags.Tag
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestPart
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.multipart.MultipartFile

@RestController
@Tag(name = "09-Image", description = "Image API")
@RequestMapping("/api/v2/image")
class ImageRestController(
    private val storageService: StorageService,
) {
    @PostMapping(consumes = [MediaType.MULTIPART_FORM_DATA_VALUE])
    fun uploadImage(
        @RequestPart("image") image: MultipartFile
    ): ImageUrlResponse {
        return ImageUrlResponse(
            imageUrl = storageService.save(image)
        )
    }

    @Schema(description = "Image 업로드 응답")
    data class ImageUrlResponse(
        @field:Schema(description = "이미지 URL", example = "https://example.com/image.jpg")
        val imageUrl: String
    )

}
